main.go

```diff
@@ -117,9 +125,9 @@ func (c *Controller) handleNewCSR(key string) error {
 		return fmt.Errorf("failed to list machines: %v", err)
 	}
 
-	maxPending := getMaxPending(machines.Items)
-	if pending := recentlyPendingCSRs(c.indexer); pending > maxPending {
-		klog.Errorf("Pending CSRs: %d; Max pending allowed: %d. Difference between pending CSRs and machines > %v. Ignoring all CSRs as too many recent pending CSRs seen", pending, maxPending, maxPendingCSRs)
+	maxPendingCSRs = getMaxPending(machines.Items)
```

Comment:
There's a potential data race since `maxPendingCSRs` is read during the metrics scrape which happens asynchronously. You need [SwapInt64](https://golang.org/pkg/sync/atomic/#SwapInt64)/[LoadInt64](https://golang.org/pkg/sync/atomic/#LoadInt64) (or variants) here.